File: c#/Solutions/Day1.cs
public class Day1
{
    public IEnumerable<int> GetData(string filename)
    {
        var data = File.ReadAllText(filename)
            .Split(Environment.NewLine + Environment.NewLine) // split by empty line
            .Select(x => x.Split(Environment.NewLine)); // split by line;

        return data.Select(x => x.Sum(y => int.Parse(y))); // sum all ints in groups
    }

    public int Solve(string filename)
    {
        return GetData(filename)
            .Max(); // max
    }

    public int Solve2(string filename)
    {
        var ordered = GetData(filename)
            .OrderByDescending(x => x); // order

        return ordered.Take(3).Sum(); // top 3
    }
}

File: python/solutions/day1.py
def getData(filename):
    with open(filename, "rt") as file: # close when done (e.g. c# using)
        groups = file.read().split('\n\n')
        return [sum(map(int, group.split('\n'))) for group in groups] # sum function applied to each in group, that converts each to int first

def solve(filename):
    data = getData(filename)
    return max(data)

def solve2(filename):
    data = getData(filename)
    order = sorted(data) # asc!
    top3 = order[-3:] # get the last three as a new array, can also be written as order[-3:len(order)], negative indexing + start/end indexing
    return sum(top3)
